bindings/python/src/manifest.rs

```diff
@@ -0,0 +1,355 @@
+    #[getter]
+    fn partition_spec_id(&self) -> i32 {
+        self.inner.partition_spec_id
+    }
+
+    #[getter]
+    fn content(&self) -> i32 {
+        self.inner.content as i32
+    }
+
+    #[getter]
+    fn sequence_number(&self) -> i64 {
+        self.inner.sequence_number
+    }
+
+    #[getter]
+    fn min_sequence_number(&self) -> i64 {
+        self.inner.min_sequence_number
+    }
+
+    #[getter]
+    fn added_snapshot_id(&self) -> i64 {
+        self.inner.added_snapshot_id
+    }
+
+    #[getter]
+    fn added_files_count(&self) -> Option<u32> {
+        self.inner.added_files_count
+    }
+
+    #[getter]
+    fn existing_files_count(&self) -> Option<u32> {
+        self.inner.existing_files_count
+    }
+
+    #[getter]
+    fn deleted_files_count(&self) -> Option<u32> {
+        self.inner.deleted_files_count
+    }
+
+    #[getter]
+    fn added_rows_count(&self) -> Option<u64> {
+        self.inner.added_rows_count
+    }
+
+    #[getter]
+    fn existing_rows_count(&self) -> Option<u64> {
+        self.inner.existing_rows_count
+    }
+
+    #[getter]
+    fn deleted_rows_count(&self) -> Option<u64> {
+        self.inner.deleted_rows_count
+    }
+
+    #[getter]
+    fn partitions(&self) -> Vec<PyFieldSummary> {
+        self.inner
+            .partitions
+            .clone()
+            .unwrap()
+            .iter()
+            .map(|s| PyFieldSummary { inner: s.clone() })
+            .collect()
+    }
+
+    #[getter]
+    fn key_metadata(&self) -> Vec<u8> {
+        self.inner.key_metadata.clone()
+    }
+}
+
+#[pyclass]
+pub struct PyManifestEntry {
+    inner: Arc<ManifestEntry>,
+}
+
+#[pymethods]
+impl PyManifestEntry {
+    #[getter]
+    fn status(&self) -> i32 {
+        ManifestStatus::Existing as i32
+    }
+
+    #[getter]
+    fn snapshot_id(&self) -> Option<i64> {
+        self.inner.snapshot_id
+    }
+
+    #[getter]
+    fn sequence_number(&self) -> Option<i64> {
+        self.inner.sequence_number
+    }
+
+    #[getter]
+    fn file_sequence_number(&self) -> Option<i64> {
+        self.inner.file_sequence_number
+    }
+
+    #[getter]
+    fn data_file(&self) -> PyDataFile {
+        PyDataFile {
+            inner: self.inner.data_file.clone(),
+        }
+    }
+}
+
+#[pyfunction]
+pub fn read_manifest_entries(bs: &[u8]) -> PyManifest {
+    // TODO: Some error handling
+    PyManifest {
+        inner: Manifest::parse_avro(bs).unwrap(),
+    }
+}
+
+#[pyclass]
+pub struct PyManifestList {
+    inner: ManifestList,
+}
+
+#[pymethods]
+impl crate::manifest::PyManifestList {
+    fn entries(&self) -> Vec<PyManifestFile> {
+        self.inner
+            .entries()
+            .iter()
+            .map(|file| PyManifestFile {
+                inner: file.clone(),
+            })
+            .collect()
+    }
+}
+
+#[pyfunction]
+pub fn read_manifest_list(bs: &[u8]) -> PyManifestList {
+    PyManifestList {
+        inner: ManifestList::parse_with_version(bs, FormatVersion::V2).unwrap(),
```

Comment:
fine for now, but this limits us to only reading V2 manifest list

Reply:
In PyIceberg we project everything to V2. I think that would be the easiest for Rust as well. I've created an issue: https://github.com/apache/iceberg-rust/issues/1471